### ISSUE TITLE
Support pino logger customAttributeKeys to replace existing req res reqId log record fields

### ISSUE THREAD

**Issue by p-fedyukovich** [fastify/fastify#4413]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

It would be good to support `customAttributeKeys` as [pino-http](https://github.com/pinojs/pino-http) does. Currently there is no elegant way to change for example `req` key to `httpRequest`

### Motivation

Google cloud log entry has own logging structure, including [`httpRequest`](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest). By adding custom attribute key `req: 'httpRequest'` and own custom request serializer it would be possible easily configure logger for google cloud log structure. 

### Example

Example for google cloud: 
```typescript    
private readonly server: FastifyInstance = fastify({
        trustProxy: true,
        ignoreTrailingSlash: true,
        logger: {
            // @ts-ignore
            customAttributeKeys: {
                req: 'httpRequest',
                reqId: 'logging.googleapis.com/trace',
            },
       }
})
```

**Comment by mcollina** [comment]
Thanks for reporting! Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by p-fedyukovich** [comment]
> Thanks for reporting! Would you like to send a Pull Request to address this issue? Remember to add unit tests.

Thanks for your reply. Yea, sure

**Comment by minhtriet359** [comment]
I'm working on this.

**Comment by climba03003** [comment]
Shouldn't it be live inside `pino` instead of here?
Or even a custom transportor?

**Comment by RafaelMeir4** [comment]
Hi! @mcollina @Eomm 

I'd like to work on this feature request. I noticed that PR(5855) by minhtriet359 was closed, and I've analyzed the issues with the previous implementation approach.

**Questions:**
1. Is this feature still available for contribution?
2. Can someone from the team confirm the preferred implementation approach? I have some ideas but would like guidance before starting.

I've studied the codebase and understand why the previous PR had compatibility issues. I believe I have a better approach in mind that would integrate more naturally with Fastify's existing logging architecture.

This would be my first contribution to Fastify, so I'd appreciate any guidance from the maintainers before I start working on it.

Thanks!

**Comment by mcollina** [comment]
I can do my best, but time is what it is - I try to review all PRs.

**Comment by RafaelMeir4** [comment]
@mcollina 

I've implemented the `customAttributeKeys` feature requested in #4413. 

This allows users to customize the log attribute keys (`req`, `res`, `err`).

When you have time, I'd appreciate a review. Thanks!

**Comment by smith558** [comment]
Hi all! @mcollina @Eomm I was thinking of doing this, but realised this is mostly done, and the issue should be **closed**, as superseded by #6580.

Fastify does not support `pino-http`’s `customAttributeKeys` option directly, but the new `LogController` (https://fastify.dev/docs/latest/Reference/Server/#logcontroller) provides a more general solution: internal log methods can be overridden to rename `req`, `res`, and other fields, while `requestIdLogLabel` handles `reqId`.

This addresses the original @p-fedyukovich's GCP logging use case. Unless direct `customAttributeKeys` compatibility is still desired, the issue appears resolved.

**Comment by climba03003** [comment]
Closing https://github.com/fastify/fastify/pull/6580
